### PR TITLE
[LAE-6221]: Add support for `message` field

### PR DESCRIPTION
**Pull request by subham-deepsource**
### Fixes and improvements
- Consistent logs (adjust spacing, fix messages, etc.)
- Error message should not end with punctuations
- Log errors to `stderr` instead of `stdout`
- Fix (flaky) tests and make them better
- Fix return code in case there's an error
- Move away from deprecated `ioutil` to `os` or `io`, etc.

### New features
- Add support for `message` field

Sample run on subham-deepsource/demo-go:
- Before
```txt
DeepSource | Artifact published successfully

Analyzer  test-coverage
Key       go
```

- After
```txt
DeepSource | Artifact published successfully

Analyzer  test-coverage
Key       go
Message   Artifact successfully uploaded for repository demo-go on commit a65448d40ef9159d7003a49c866cf53aa75bac5e.
```

### CODE REVIEW

File: command/report/report.go [hunk 1]
```diff
@@ -256,20 +252,14 @@ func (opts *ReportOptions) Run() int {
 	// }
 
 	if !queryResponse.Data.CreateArtifact.Ok {
-		fmt.Println("DeepSource | Error | Reporting failed | ", queryResponse.Data.CreateArtifact.Error)
+		fmt.Fprintln(os.Stderr, "DeepSource | Error | Reporting failed |", queryResponse.Data.CreateArtifact.Error)
 		sentry.CaptureException(errors.New(queryResponse.Data.CreateArtifact.Error))
 		return 0
```
Comment (siddhant-deepsource) [inline command/report/report.go:257]:
```suggestion
		return 1
```

Comment (siddhant-deepsource) [reply command/report/report.go:257]: @subham-deepsource there are some other instances above where we return `0` here. Ideally we should return `1` in those failure cases. Can you please make that change?

**Review by siddhant-deepsource — CHANGES_REQUESTED** [review]

File: command/report/report.go (hunk 1 above)
Comment (subham-deepsource) [reply command/report/report.go:257]: Yeah, right.

**Comment by siddhant-deepsource** [comment]
@subham-deepsource test is failing.

**Comment by subham-deepsource** [comment]
> @subham-deepsource test is failing.

Fixed it.

**Review by siddhant-deepsource — APPROVED** [review]

**Review by siddhant-deepsource — APPROVED** [review]

**Review by sourya-deepsource — APPROVED** [review]

**Pull request merged**